### ISSUE TITLE
[Enhancement] Invert default value of xct:TouchEffect.ShouldMakeChildrenInputTransparent

### ISSUE THREAD

**Issue by Tommigun1980** [xamarin/XamarinCommunityToolkit#852]
## Summary

Adding an `xct:TouchEffect.NativeAnimation="True"` in your visual hierarchy prevents touch events from working inside said element. This forces one to also always add an `xct:TouchEffect.ShouldMakeChildrenInputTransparent="False"` statement when a 'NativeAnimation' statement is added.

This is error prone, cumbersome, and may scare off new users as their app's logic will break without realising this has to be done.

## API Changes

Change `xct:TouchEffect.ShouldMakeChildrenInputTransparent`'s default value to "False" instead of "True".

Blocking child input is imho definitely a feature that should be **opt in** and not opt out.

## Who Will Do The Work?

- [ ] I am willing to take this on myself
- [X] Just putting this out there for someone to pick up

**Comment by jfversluis** [comment]
That makes sense. @AndreiMisiukevich what do you think? Was this done with a certain reason behind it?

This is a breaking change though, so we should definitely target our vNext version.

**Comment by AndreiMisiukevich** [comment]
> That makes sense. @AndreiMisiukevich what do you think? Was this done with a certain reason behind it?
> 
> This is a breaking change though, so we should definitely target our vNext version.

I definitely did it for some reason :)
But I don't remember it LoL

**Comment by AndreiMisiukevich** [comment]
Found it.
I did it because of the Android issue. (I will make a recording)

**Comment by AndreiMisiukevich** [comment]
https://user-images.githubusercontent.com/10124814/107285408-668ebd00-6a5f-11eb-9da1-b6b18d6aa96d.mp4

**Comment by AndreiMisiukevich** [comment]
@Tommigun1980 if you wish, I can add ```bool? DefaultShouldMakeChildrenInputTransparent``` property for setting this value only once.

**Comment by AndreiMisiukevich** [comment]
Closing this issue because otherwise, unfortunately, it won't work fine on Android (at least on Android)

**Comment by Tommigun1980** [comment]
@AndreiMisiukevich Does this mean that Android isn’t compatible with the effect then?
I mean you have to set the flag to False in order not to break input, and you’re saying setting it to False doesn’t work on Android?
Shouldn’t this be fixed?
Thank you.

**Comment by AndreiMisiukevich** [comment]
@Tommigun1980 I don't think that Android isn’t compatible with the effect.

This effect allows creating complex touchable layouts.

Of course, it would be great if we could fix it, but I am not sure that it's possible. (By default XF views on an android intercept gestures as you can see on the video).

**Comment by Tommigun1980** [comment]
Any reason the default can't be not to block child input if it works on
Android? I'm not sure I'm seeing the point.
The default to block input breaks most controls. If it can be turned off on
Android why not make it the default? If it can't be turned off on Android
it's going to be very hard to use, no?
All I want to use it for is to add a touch highlight on arbitrary controls.
Thank you again.

**Comment by AndreiMisiukevich** [comment]
@Tommigun1980 please check the video I attached above.
The problem is that it doesn't work on Android if we don't block children's input.

As you can see on the video, the control's state remains "pressed" even if touch is released in case the touch was started on some child inside the "touchable" view.

**Comment by jpozo20** [comment]
@AndreiMisiukevich seems there is some issue on IOS too. 

As you said, on Android the view stays in the "pressed" state if you do a press and move the cursor out of that view.
But when testing on iOS I found the parent view is receiving the click even though `ShouldMakeChildrenInputTransparent` is set to false.

I have a Grid → Grid → Switch. The command is set on the first grid and when you click the Switch the Grid's command is executed.

**Comment by AndreiMisiukevich** [comment]
@jpozo20 yeah. I don't know how to fix that to have it as "common" solution.

**Comment by jpozo20** [comment]
@AndreiMisiukevich I feel you, I was reading the code to understand how to replicate it and yeah it's a bit complicated. 

I managed to do the ripple animation by creating an Effect and letting Android do the animation with a StateListDrawable. But seeing how much trial and error I had to do, I don't see it very feasible to modify the current code in the TouchEffect since it seems you're actively managing the animation of the drawable (maybe due to the available properties?).
